### PR TITLE
AP-3768: Improve maintenance page

### PR DESCRIPTION
**Pull request by jsugarman**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3768)

Rushed worked to put a maintenance page up was not ideal.

1. move maintenance mode to config from `Setting` model
2. add request specs or other integration spec for coverage
3. enable per environment configuration
4. add documentation

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/requests/pages_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
+require "rails_helper"
+
+RSpec.describe PagesController, clamav: true do
```
Comment (colinbruce) [inline spec/requests/pages_controller_spec.rb:3]: As you've added the tests, can you remove the simplecov exclusion? https://github.com/ministryofjustice/laa-apply-for-legal-aid/blame/main/.simplecov#L10

Comment (jsugarman) [reply spec/requests/pages_controller_spec.rb:3]: oh, i thought i had. will do.

Comment (jsugarman) [reply spec/requests/pages_controller_spec.rb:3]: yeah, i had. https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/6100/files#diff-24002d905f0809599eae02b5cd7b58f8821aa89272b2b207c3e589b7a7a4f3a6L10


will remove first commit now though

Comment (colinbruce) [reply spec/requests/pages_controller_spec.rb:3]: Huh! How did I miss that! 🤦‍♂️

**Review by colinbruce — APPROVED** [review]
I still have concerns about changes being a two-dev solution but a good iteration over version 1 :+1:

**Pull request merged**